test/framework/fileConversion.go

```diff
@@ -1,33 +1,32 @@
 package framework
 
 import (
-	"bytes"
-	"compress/gzip"
 	"fmt"
+	"os"
+	"os/exec"
 
 	"github.com/kubevirt/containerized-data-importer/pkg/image"
 )
 
-// createGz compresses a slice of data in gzip format
-func CreateGz(src []byte) ([]byte, error) {
-
-	var buf bytes.Buffer
-
-	gzw := gzip.NewWriter(&buf)
-	defer gzw.Close()
-
-	n, err := gzw.Write(src)
-	if n != len(src) {
-		return nil, fmt.Errorf("CreateGz: source size (%d) does not match bytes written (%d)\n", n, len(src))
+// CreateGz takes the path of a file and stats prior to calling the OS's gzip binary
+// It returns the new names of the file or an error
+func CreateGz(srcFile string) (string, error) {
```

Comment:
can creation of various formats be _table-ized_, where the table contains the ext, the cmd and args needed to create the format and maybe an output path?